### DISCUSSION TITLE
Problem after deletion of Data Object Class with remaining data object entries (Docker container context)

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#16213] [Q&A / Support]
Hi,

I have the following problem:
After I delete a data object class I'm not able to delete remaining data object entries, because it requires the class to be present. I think this was different in an early Pimcore version?

Sure I could delete all class entries before I delete the class definition itself. But this doesn't work in a multi environment or with multiple developers. Is it ment to be like it is currenly?

Is there a way to delete the remaining data object from the db? Is there a console command to use? Or would I have to do it directly in the db? If so, which sql commands would I have to execute?

Thanks in advance for any help!

**Comment by dvesh3** [comment]
AFAIK it should cleanup the objects before deleting the class definition. please see https://github.com/pimcore/pimcore/blob/aed88d9d2604c6c2d153f5eef98d5c3e5f95819e/models/DataObject/ClassDefinition.php#L490-L497

If there are still some objects in the db, then i would check the logs if there was an error deleting these objects.

**Reply by akno77** [reply]
@dvesh3  This might work if you are the one, who deletes the class in the Pimcore admin, but this doesn't work if the class definition file is removed from the Git repo and the code is deployed to the servers via "bin/console pimcore:deployment:classes-rebuild" command. Or do I miss sth?

**Reply by dvesh3** [reply]
then you need to run the command with `delete-classes` option `bin/console pimcore:deployment:classes-rebuild -d` to perform cleanup for deleted definitions.

**Reply by akno77** [reply]
@dvesh3 you are right, this works in most cases! But I forgot to mention that we deploy docker containers via GitLab CI, so the source is build from scratch which means there aren't any Class files left in "var/classes/DataObject" for the deleted defintions. So the command `bin/console pimcore:deployment:classes-rebuild -d` throws an exception saying the Class file couldn't be found...
